### PR TITLE
Docs: Extended toolbar guide with multiline introduction

### PR DESCRIPTION
**Pull request by godai78**
Closes: #8636.

New toolbar guide separated from the main configuration guide. New demos for different multiline behavior; extended description; updated crosslinks.

New multiline toolbar behaviour introduced in selected demos: arbitrary break point set in Autoformat and Paste from guides.

### CODE REVIEW

File: docs/builds/guides/integration/toolbar-setup.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    toolbar: [ 'bold', 'italic', '|', 'undo', 'redo' ]
+```
+
+```js
+    toolbar: {
+        items: [ 'bold', 'italic', '|', 'undo', 'redo' ]
+    }
+```
+
+It is also possible to arrange toolbar items into multiple lines. In the extended format set `shouldNotGroupWhenFull` option to `true`, so items will not be grouped when the toolbar overflows but will wrap to the new line instead. Additionally, a `'-'` could be used inside items list to set the breaking point explicitly.
```
Comment (pkwasnik) [inline docs/builds/guides/integration/toolbar-setup.md:59]: I am considering adding here additional header, e.g.: `#### Multiple lines` to enhance this feature a bit more.

**Review by pkwasnik — APPROVED** [review]

File: docs/builds/guides/integration/toolbar-setup.md (hunk 1 above)
Comment (godai78) [reply docs/builds/guides/integration/toolbar-setup.md:61]: Added a headline, changed the headlines to reflect taking it out from configuration.

File: docs/_snippets/builds/guides/integration/toolbar-breakpoint.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<div id="toolbar-breakpoint">
+
+<h2>The nice Canadian</h2>
+<p>Remember Chris Hadfield? Of course, everyone has seen &ldquo;Space Oddity&rdquo; played and sung onboard the ISS. Have you seen the videos he recorded for schools? Short pieces on living in space, brushing your teeth in space, and sleeping in zero gravity? Kids love them. Teachers love them. Schools do use them. This is a gift, a thank you gesture, a hand extended back. This humble man made his dream come true, he flew to the stars. But not even once did he forget the people that put him there &ndash; the engineers, designers, scientists.&nbsp;Those before him that went into space and those who gave their life to make this dream possible.</p>
```
Comment (AnnaTomanek) [inline docs/_snippets/builds/guides/integration/toolbar-breakpoint.html:4]:
```suggestion
<p>Remember Chris Hadfield? Of course, everyone has seen &ldquo;Space Oddity&rdquo; played and sung onboard the ISS. Have you seen the videos he recorded for schools? Short pieces on living in space, brushing your teeth in space, and sleeping in zero gravity? Kids love them. Teachers love them. Schools use them. This is a gift, a thank you gesture, a hand extended back. This humble man made his dream come true, he flew to the stars. But not even once did he forget the people that put him there &ndash; the engineers, designers, scientists.&nbsp;Those before him that went into space and those who gave their life to make this dream possible.</p>
```

File: docs/_snippets/builds/guides/integration/toolbar-breakpoint.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+<p>Now he helps bring up a new generation that will build better, safer, faster spaceships. That will once more reach the surface of Moon, will leave the Earth orbit and fly to Mars, to Venus. Those are years to come and the goals will take a lot of patience. But what he does is gold. He is honest, humble, and true in what he did up there and the generation that came can see that, grab that passion, can share in his activities.</p>
```
Comment (AnnaTomanek) [inline docs/_snippets/builds/guides/integration/toolbar-breakpoint.html:6]:
```suggestion
<p>Now he helps bring up a new generation that will build better, safer, faster spaceships. That will once more reach the surface of the Moon, will leave the Earth orbit and fly to Mars, to Venus. Those are years to come and the goals will take a lot of patience. But what he does is gold. He is honest, humble, and true in what he did up there and the generation that came can see that, share in his activities and grab that passion.</p>
```

File: docs/_snippets/builds/guides/integration/toolbar-grouping.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<div id="toolbar-grouping">
+
+<h2>The space race</h2>
+
+<p>It is no longer a secret &ndash; though it was for a long time &ndash; that the Soviet Union raced the United States to the Moon and only in the knowledge of their own failure did they focus all their brilliant minds and hard work on the orbital station program known widely as Salyut. The reason for their failure, we learned in the late 1990&rsquo;s, was the lack of a proper plan. And, as some sources suggest, the absence of a real need to achieve the goal. It was the US president&rsquo;s challenge, cast in front of an audience, that the Americans will put a man on the face of Luna before the end of the decade.</p>
```
Comment (AnnaTomanek) [inline docs/_snippets/builds/guides/integration/toolbar-grouping.html:5]:
```suggestion
<p>It is no longer a secret &mdash; though it was for a long time &mdash; that the Soviet Union raced the United States to the Moon and only in the knowledge of their own failure did they focus all their brilliant minds and hard work on the orbital station program known widely as Salyut. The reason for their failure, we learned in the late 1990&rsquo;s, was the lack of a proper plan. And, as some sources suggest, the absence of a real need to achieve the goal. It was the US president&rsquo;s challenge, cast in front of an audience, that the Americans will put a man on the face of Luna before the end of the decade.</p>
```

File: docs/_snippets/builds/guides/integration/toolbar-grouping.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+<p>The Soviets, who have so far been winning in everything concerning the space race, had to accept the challenge. They beat the Americans with the first manned spaceflight, the first (and for almost twenty more years only) woman in space, the first multicraft flight, and finally &ndash; the first group flight. They left Americans on scorched ground, even if sometimes just by weeks or even days. And in the course of racing the hideous imperialist capitalist to the Silver Globe they achieved one more outstanding success &ndash; the first orbital docking between two manned space crafts and a successful orbital crew transfer.</p>
```
Comment (AnnaTomanek) [inline docs/_snippets/builds/guides/integration/toolbar-grouping.html:7]:
```suggestion
<p>The Soviets, who have so far been winning in everything concerning the space race, had to accept the challenge. They beat the Americans with the first manned spaceflight, the first (and for almost twenty more years only) woman in space, the first multicraft flight, and finally &mdash; the first group flight. They left Americans on scorched ground, even if sometimes just by weeks or even days. And in the course of racing the hideous imperialist capitalist to the Silver Globe they achieved one more outstanding success &mdash; the first orbital docking between two manned spacecrafts and a successful orbital crew transfer.</p>
```

**Review by AnnaTomanek — COMMENTED** [review]
Some other suggestions added.

File: packages/ckeditor5-autoformat/docs/_snippets/features/autoformat.js [hunk 6]
```diff
@@ -35,6 +36,7 @@ ClassicEditor
 				'undo',
 				'redo'
 			],
+			shouldNotGroupWhenFull: true,
```
[the remark this answers is not in the document]
Comment (godai78) [reply packages/ckeditor5-autoformat/docs/_snippets/features/autoformat.js:39]: Removed.

**Comment by AnnaTomanek** [comment]
TODO: Bring back TOC in guides with multiline toolbar that had it removed.

**Comment by godai78** [comment]
> TODO: Bring back TOC in guides with multiline toolbar that had it removed.

These were removed in the Pagination / Export to guides for a different reason I believe. I have not removed any ToC in no guide whatsoever ;)

**Comment by godai78** [comment]
@ckeditor/qa-team - guys, I had to make some changes in the snippets, can you just look it over? It works fine, but maybe I missed some conventions that should be kept.

File: packages/ckeditor5-paste-from-office/docs/_snippets/features/paste-from-office.js [hunk 7]
```diff
@@ -23,7 +23,7 @@ ClassicEditor
 				'italic',
 				'underline',
 				'strikethrough',
-				'|',
+				'-',
```
[the remark this answers is not in the document]
Comment (godai78) [reply packages/ckeditor5-paste-from-office/docs/_snippets/features/paste-from-office.js:26]: A few more items added to balance the look better.

**Review by AnnaTomanek — APPROVED** [review]
Looks good to me. Congratulations on creating and updating so many snippets! 👏

**Comment by godai78** [comment]
> Looks good to me. Congratulations on creating and updating so many snippets! 👏

This was quite a battle, I need to admit. Especially when I discovered one toolbar item was dead.

**Pull request merged**